lib/rspec/mocks/space.rb

```diff
@@ -164,6 +171,24 @@ def proxy_not_found_for(id, object)
                       end
       end
 
+      def superclass_proxy_not_found_for(id, object)
+        raise "superclass_proxy_not_found_for called with something that is not a class" unless Class === object
+        proxies[id] = class_proxy_with_callback_verification_strategy(object, NoCallbackInvocationStrategy.new)
+      end
+
+      def class_proxy_with_callback_verification_strategy(object, strategy)
+        if RSpec::Mocks.configuration.verify_partial_doubles?
+          VerifyingPartialClassDoubleProxy.new(
+            self,
+            object,
+            @expectation_ordering,
+            strategy
+          )
+        else
+          PartialClassDoubleProxy.new(self, object, @expectation_ordering)
+        end
+      end
```

Comment:
Could we remove this check? Instead we set the proxy adapter on the configuration when `verify_partial_doubles` is configured. Something like:

```ruby
def verify_partial_doubles=(flag)
  # set the state
  if flag
    self.partial_class_double_proxy = VerifyingPartialClassDoubleProxy
  else
    self.partial_class_double_proxy = PartialClassDoubleProxy
  end
end
```
Turning section into:

```ruby
def class_proxy_with_callback_verification_strategy(object, strategy)
  RSpec::Mocks.configuration.partial_class_double_proxy.new(
    self,
    object,
    @expectation_ordering,
    strategy
  )
end
```

Or even making that a factory method on the configuration so we do not have to call `new`.